libs/core/src/components/sage-button/sage-button.tsx

```diff
@@ -0,0 +1,93 @@
+import { Component, Element, Host, h, Prop } from '@stencil/core';
+import { hasShadowDom } from '../../utils/utils';
+
+@Component({
+  tag: 'sage-button',
+  styleUrl: 'sage-button.scss',
+  shadow: true,
+})
+export class SageButton {
+  @Element() el: HTMLSageButtonElement;
+
+  /**
+   * Sets button variant styles as outlined in Figma documentation
+    */
+  @Prop() variant: 'primary' | 'secondary' | 'accent' | 'disclosure' | 'destructive' = 'primary';
+
+  /**
+   * Displays icon before text when icon string matches an icon name
+   */
+  @Prop() icon?: string = null;
+
+  /**
+   * Toggles disabled state of button
+   * @defaultValue false
+   */
+  @Prop() disabled? = false;
+
+  /**
+   * Provides button with a submittable name
+   */
+  @Prop() name?: string;
+
+  /**
+   * Provides button with a submittable value
+   */
+  @Prop() value?: string;
+
+  /**
+   * Provides button with a type 
+   * todo: make sure there is info in the story
+   * @defaultValue button
+   */
+  @Prop() type?: 'button' | 'reset' | 'submit' = 'button';
+
+
+  // todo build out e2e test to confirm form submission 
+  private handleClick = (ev: Event) => {
+    if (this.type != 'button') {
+      // If button clicked IS NOT associated with a form
+      if (hasShadowDom(this.el)) {
+        const form = this.el.closest('form')
+        if (form) {
+          ev.preventDefault()
+
+          const fakeButton = document.createElement('button')
+          fakeButton.type = this.type
+          fakeButton.style.display = 'none'
+          form.appendChild(fakeButton)
+          fakeButton.click()
+          fakeButton.remove()
+        }
+      }
+    }
+  }
+  
+  private buttonClassNames = () => {
+    let className = `sage-button`;
+    if (this.variant && this.variant != 'primary') {
+      const variantClassName = `sage-button--${this.variant}`;
+      className += ' ' + variantClassName;
+    }
+    return className;
+  };
+
+  render() {
+    const trashIcon = (
```

Comment:
Something about this icon is throwing off the button alignment. It's extremely subtle, but if you squint you notice it in the stories.